Hw2/Task2/MatrixTracker.cs

```diff
@@ -0,0 +1,31 @@
+namespace Matrix
+{
+    public class MatrixTracker<T>
+    {
+        private Stack<ElementChangedEventArgs<T>> changes = new();
+        private DiagonalMatrix<T> matrix;
+        public MatrixTracker(DiagonalMatrix<T> matrix)
+        {
+            matrix.ElementChanged += OnMatrixChanged;
+            this.matrix = matrix;
+        }
+
+        private void OnMatrixChanged(object sender, ElementChangedEventArgs<T> e)
+        {
+            changes.Push(e);
+            // Console.WriteLine("Change: " + string.Join(" ", [e.I, e.J, e.OldValue, e.NewValue]));
+        }
+
+        public void Undo()
+        {
+            if(changes.Count == 0)
+            {
+                return;
+            }
+            ElementChangedEventArgs<T> change = changes.Pop();
```

Comment:
Unsubscribe method  OnMatrixChanged, to avoid loop, and subcribe it again after change is made